processor/src/main/java/org/mapstruct/ap/internal/processor/DefaultVersionInformation.java

```diff
@@ -92,6 +95,14 @@ static DefaultVersionInformation fromProcessingEnvironment(ProcessingEnvironment
         String runtimeVersion = System.getProperty( "java.version" );
         String runtimeVendor = System.getProperty( "java.vendor" );
 
+        String javaSpecificationVersion = System.getProperty( "java.specification.version" );
+
+        if ( javaSpecificationVersion.equals( "1.6" ) || javaSpecificationVersion.equals( "1.7" ) ) {
```

Comment:
what about even earlier versions, 1.5 introduced annotation processing.  possibly not needed..

Reply:
I am not 100% convinced with using the java.specification property for this. I think that there are better ways for this and also the place for doing this check is not OK. Thinking more about this I think it is not that crucial to have it

Reply:
>  I think it is not that crucial to have it
Maybe you are right. We haven't bugs about it, so... I can continue to work with it. I mean to fiend better way and better place if we need it. I'm waiting for your decision :)